### ISSUE TITLE
Enable VNC at boot

### ISSUE THREAD

**Issue by hamzamerzic** [maxnet/berryboot#175]
Hello. I am trying to make vnc server start at berryboot boot for couple of days now, but I can't seem to find a way. I am trying to make it possible to remotely choose which OS to boot, as I have Debian and OpenElec installed. My cmdline.txt is as fallows:

```
datadev=mmcblk0p2 smsc95xx.turbo_mode=N elevator=deadline quiet bootmenutimeout=20
sound=hdmi vncinstall ipv4=192.168.1.24/255.255.255.0/192.168.1.1/wlan0
```

It works great without the last two commands, but when I define vncinstall and ipv4 I get the fallowing message:

```
Successfully initialized wpa_supplicant
rfkill: Cannot open RFKILL control device
Could not read interface wlan0 flags: No such device
WEXT: Could not set interface 'wlan0' UP
wlan0: Failed to initialize driver interface
numid=3,iface='MIXER',name='PCM Playback Route'
    :  type=INTEGER,access=rw------,values=1,min=0,max=2,step=0
    :  values=2
```

I have setup wlan inside berryboot, so my wpa_supplicant is working. I suspect that the problem could be that there is not enough time to load my wifi usb dongle, as (if I understood the code correctly) there are only 2 seconds for wifi checking (Installer::startWifi()).

I have managed to start it once. I don't know how or why, but I didn't get the message and was able to use vnc to select the desired OS to boot. Perhaps that time it was able to find the dongle and turn it on during those two second, I'm not sure.

If you have any advice for fixing this I would be grateful. Thank you for time and for building this awesome utility.

**Comment by WilsonBradley** [comment]
It seems when I played around with vncinstall option, it seems it was only for berryboot. I got it to have the BerryMenu in VNC but once you start a distro, it disconnects. I think that is the nature of it.

**Comment by hamzamerzic** [comment]
Yes, I am aware of that. As I said: "I am trying to make it possible to remotely choose which OS to boot...", so I just need vnc inside berryboot, but it gives me the mentioned errors. Thank you for your insight though.

**Comment by maxnet** [comment]
> there are only 2 seconds for wifi checkin

If you type "dmesg", does the timing information reveal how long your dongle takes to get ready?

Prefer not to wait too long as it is not certain a wifi dongle is connected at all.
User may have used it only once and unplugged it afterwards.

**Comment by hamzamerzic** [comment]
As I suspected, it takes more than 2 seconds to load the wifi dongle (namely 3.54 in my case). Thank you for your help, I will try rebuilding berryboot with wait period set to 4 seconds and post here how it went.

EDIT: OK, so I have changed the time to 4 seconds and built berryboot. It worked great during headless install, as I was able to start the vnc client without any problems. But after I finished the installation and made a reboot I got the same error as stated. Maybe there is a flag which makes berryboot behave differently after initial configuration?
